### PR TITLE
Update Command_nickclean.java

### PR DESCRIPTION
**Pull request by Taahh**
https://totalfreedom.boards.net/thread/66931/nickclean-1-person per this suggestion, an option to switch /nickclean to /nickclean <player | -a>

sorry i've never done this before and i already went in game and tested it, works.

### CODE REVIEW

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_nickclean.java [hunk 1]
```diff
@@ -23,12 +23,39 @@
     @Override
     public boolean run(CommandSender sender, Player playerSender, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
     {
-        FUtil.adminAction(sender.getName(), "Cleaning all nicknames", false);
-
-        for (final Player player : server.getOnlinePlayers())
+        if (args.length == 0)
```
Comment (ncurran02) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_nickclean.java]: Why not `args.length != 1`, the command only requires one argument (player name or `-a`)

**Comment by VideoGameSmash12** [comment]
sorry for being nitpicky but i think you should update the description of the command to reflect the changes made here

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_nickclean.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 import org.bukkit.entity.Player;
 
 @CommandPermissions(level = Rank.SUPER_ADMIN, source = SourceType.BOTH)
-@CommandParameters(description = "Essentials Interface Command - Remove illegal chatcodes from nicknames of all players on server.", usage = "/<command>", aliases = "nc")
+@CommandParameters(description = "Essentials Interface Command - Remove illegal chatcodes from nicknames of all players on server.", usage = "/<command> <player | -a>", aliases = "nc")
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_nickclean.java:15]: No longer is accurately describing what the command does.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_nickclean.java [hunk 3]
```diff
@@ -23,12 +23,39 @@
     @Override
     public boolean run(CommandSender sender, Player playerSender, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
     {
-        FUtil.adminAction(sender.getName(), "Cleaning all nicknames", false);
-
-        for (final Player player : server.getOnlinePlayers())
+        if (args.length != 1)
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_nickclean.java:26]: Any reason not to default to current behavior?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_nickclean.java [hunk 4]
```diff
@@ -23,12 +23,39 @@
...
+        } else {
+            Player player = getPlayer(args[0]);
+            if (player == null)
+            {
+                msg(PLAYER_NOT_FOUND);
+                return true;
+            }
             final String playerName = player.getName();
             final String nickName = plugin.esb.getNickname(playerName);
+            FUtil.adminAction(sender.getName(), "Cleaning " + player.getName() + "'s nickname", false);
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_nickclean.java:58]: Does this need to be broadcasted to all players?

**Review by Wild1145 — CHANGES_REQUESTED** [review]

**Comment by script-head** [comment]
Doing this differently. The way I'm doing this is you can optionally specify a specific player, but it defaults to it's original clean all nicknames.

**Pull request closed without merging**